File: app/src/main/kotlin/metamark/parser/StringLiteralTerm.kt
package metamark.parser

import metamark.parser.antlr.MetamarkParser

data class StringLiteralTerm(
    val value: String,
) : ExpressionTerm() {
    companion object {
        fun build(ctx: MetamarkParser.StringLiteralContext): StringLiteralTerm {
            val rawText = ctx.StringLiteral().text

            return StringLiteralTerm(
                value = rawText.drop(1).dropLast(1),
            )
        }
    }
}

File: src/main/StringLiteralTerm.ts
import {StringLiteralContext} from "../generated_parser/MetamarkParser";

export class StringLiteralTerm {
    static build(ctx: StringLiteralContext): StringLiteralTerm {
        const rawText = ctx.StringLiteral().text;
        return new StringLiteralTerm(rawText.substring(1, rawText.length - 1));
    }

    readonly value: string;

    constructor(value: string) {
        this.value = value;
    }
}
